File: src/main/java/org/snake/models/basic/PointSpace.java
package org.snake.models.basic;

public class PointSpace {
    private final Range xRange;
    private final Range yRange;

    public PointSpace(Range xRange, Range yRange) {
        this.xRange = xRange;
        this.yRange = yRange;
    }

    public Point around(Point point) {
        if(xRange.contained(point.x) && yRange.contained(point.y)) {
            return point;
        } else {
            return new Point(xRange.around(point.x), yRange.around(point.y));
        }
    }
}

File: src/main/kotlin/org/snake/models/basic/PointSpace.kt
package org.snake.models.basic

class PointSpace(private val xRange: Range, private val yRange: Range) {
    fun around(point: Point?): Point? {
        return if (xRange.contained(point!!.x) && yRange.contained(point.y)) {
            point
        } else {
            Point(xRange.around(point.x), yRange.around(point.y))
        }
    }
}